File: frontend/src/constants/requirementStatus.ts
import type { TagProps } from 'naive-ui';

export const REQUIREMENT_STATUS_OPTIONS = [
  { label: '草稿', value: 'draft' },
  { label: '待评审', value: 'pending_review' },
  { label: '设计中', value: 'designing' },
  { label: '开发中', value: 'developing' },
  { label: '测试中', value: 'testing' },
  { label: '待发版', value: 'pending_release' },
  { label: '已发版', value: 'released' },
  { label: '已完成', value: 'completed' },
  { label: '已关闭', value: 'closed' },
] as const;

export function requirementStatusLabel(key: string): string {
  return REQUIREMENT_STATUS_OPTIONS.find((o) => o.value === key)?.label ?? key;
}

export function requirementStatusTagType(key: string): TagProps['type'] {
  if (key === 'released' || key === 'completed') return 'success';
  if (key === 'closed') return 'default';
  if (key === 'testing' || key === 'pending_release') return 'info';
  if (key === 'developing' || key === 'designing') return 'warning';
  if (key === 'pending_review') return 'warning';
  return 'default';
}

File: backend/app/constants/requirement_status.py
from __future__ import annotations

REQUIREMENT_STATUS_KEYS: tuple[str, ...] = (
    "draft",
    "pending_review",
    "designing",
    "developing",
    "testing",
    "pending_release",
    "released",
    "completed",
    "closed",
)

REQUIREMENT_STATUS_LABELS: dict[str, str] = {
    "draft": "草稿",
    "pending_review": "待评审",
    "designing": "设计中",
    "developing": "开发中",
    "testing": "测试中",
    "pending_release": "待发版",
    "released": "已发版",
    "completed": "已完成",
    "closed": "已关闭",
}

# 手动保持的状态：不参与工作流推导
HOLD_REQUIREMENT_STATUS_KEYS: frozenset[str] = frozenset({"closed", "completed"})

# 终结态：不参与进行中待办推导
TERMINAL_REQUIREMENT_STATUS_KEYS: frozenset[str] = frozenset({"released", "completed", "closed"})

# 工作台待办：开发中
TESTER_TODO_DEVELOPING_STATUS_KEYS: tuple[str, ...] = ("developing",)

# 工作台待办：测试中 + 待发版
TESTER_TODO_TESTING_STATUS_KEYS: tuple[str, ...] = ("testing", "pending_release")


def requirement_status_label(key: str) -> str:
    return REQUIREMENT_STATUS_LABELS.get(key, key)
